pkg/preparer/podprocess/reporter.go

```diff
@@ -241,8 +271,46 @@ func (r *Reporter) initWorkspaceDir() error {
 	return nil
 }
 
-func (r *Reporter) publishProcessExits(quitCh <-chan struct{}) {
+func (r *Reporter) mainLoop(quitCh <-chan struct{}) {
 	defer r.finishService.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		r.publishProcessExits(quitCh)
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		r.pruneRows(quitCh)
```

Comment:
What happens if this doesn't return? Can we make pruning best effort? I'd feel more comfortable with this approach if we had an error reporter wired up.